pkg/reconciler/ingress/ingress.go

```diff
@@ -45,13 +46,16 @@ const (
 	privateLBDomain         = "knative-local-gateway.istio-system.svc.cluster.local"
 	producerServiceName     = "async-producer"
 	asyncOriginalHostHeader = "Async-Original-Host"
+	ingressClassName        = "INGRESS_CLASS_NAME"
 )
 
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
 	logger := logging.FromContext(ctx)
-	// TODO(bvennam): allow this ingress class to be configurable
-	ingressClass := networkpkg.IstioIngressClassName
+	ingressClass := os.Getenv(ingressClassName)
+	if ingressClass == "" {
```

Comment:
What if the value is garbage (incorrect)? You will be setting up for failure that could be hard to debug. So please be defensive here and any “garbage” value should be rejected and logged and set default. Thx.

Reply:
Debug level logging has been added to verify if a valid suffix ingress has been detected or if it was set to default.

Testing now includes base case (covers blank/incorrect suffix) and the correct suffix that is non default case (kourier).